router/types/types.go

```diff
@@ -1,88 +1,138 @@
 package router
 
 import (
-	"encoding/json"
-	"errors"
+	"fmt"
 	"time"
 )
 
+// Route is a struct that combines the fields of HTTPRoute and TCPRoute
+// for easy JSON marshaling.
 type Route struct {
-	ID        string `json:"id,omitempty"`
+	// Type is the type of Route, either "http" or "tcp".
+	Type string `json:"type"`
+	// ID is the unique ID of this route.
+	ID string `json:"id,omitempty"`
+	// ParentRef is ... TODO
```

Comment:
`ParentRef` is an external opaque identifier used by the route creator for filtering and correlation. That being said, it's kind of a hack and we should come up with a better way of doing this (the controller puts the app ID in there)